src/idiomatic/leveraging-the-type-system/raii/drop_bomb.md

```diff
@@ -0,0 +1,91 @@
+# Drop Bombs: Enforcing API Correctness
+
+Use `Drop` to enforce invariants and detect incorrect API usage. A "drop bomb"
+panics if not defused.
```

Comment:
For the members of the audience familiar with C++, consider mentioning that in C++ a private destructor is used to achieve a similar effect. So if your C++ API design intuition tells you to make the destructor not accessible to the user, use this pattern in Rust.

Reply:
Sure thing, C++ was my first primary language for many years. Wasn't always certain exactly about what languages I should refer and which not.

Reply:
From an internal (sorry!) design doc for this class:

> Target audience
>
> Engineers with at least 2-3 years of coding experience in C, C++11 or newer, Java 7 or newer, Python 2 or 3, Go or any other similar imperative programming language. We have no expectation of experience with more modern or feature-rich languages like Swift, Kotlin, C#, or TypeScript.